doc/HD44780.rst

```diff
@@ -0,0 +1,197 @@
+a significant limitation.  There can only be 8 characters worth of 'custom'
+content displayed at any point in time.  While this is quite limited, with some
+creativity it can be very useful.
+
+To understand how to leverage this capability requires some explanation of
+how the screen of the display is organized and how luma.lcd manages the
+custom character space.
+
+Each character position on an HD44780 display is made up of a 5 pixel by 8 pixel
+grid.  When a character is requested to be displayed, the HD44780 looks up the
+character from its font table and copies the pixels from the font table ROM to the
+address in RAM that corresponds to the current character position.  If there is
+not a character within the font table that matches what needs to be displayed,
+a custom character can be created within a small space in the display's RAM that
+is reserved for this purpose.  There are several restrictions that have to be
+kept in mind though.
+
+* The custom character must be a 5x8 image
+* It can only be displayed in alignment with the other characters on the display
+* There can be a maximum of 8 special characters on the screen at any given time
+
+However there are two features that can be leveraged to extend beyond these
+restrictions.
+
+* The content of a special character can be used multiple times on the screen
+* The content of all of the special characters can be changed every time the
+  screen is redrawn.
+
+Here is a small example of how this can be leveraged.
+
+.. code:: python
+
+
+  from luma.core.interface.serial import parallel
+  from luma.lcd.device import hd44780
+  from luma.core.render import canvas
+  from PIL import Image, ImageDraw
+
+  serial = parallel(RS=7, E=8, PINS=[25,24,23,27])
+  device = hd44780(serial)
+
+  def progress_bar(width, height, percentage):
+    img = Image.new('1', (width, height))
+    drw = ImageDraw.Draw(img)
+    drw.rectangle((0, 0, width-1, height-1), fill='black', outline='white')
+    drw.rectangle((0, 0, width*percentage, height-1), fill='white', outline='white')
+    return img
+
+  progress = 0.25
+  fnt = device.get_font(0)
+  with canvas(device) as draw:
+    draw.text( (5,0), f'Installing {progress*100:.0f}%', font=fnt, fill='white')
+    draw.bitmap( (5,8), progress_bar(70, 8, progress), fill='white')
+
+.. image:: images/hd44780_progress.jpg
+
+There are a few of things that deserve highlighting in this code.
+
+* We have used the hd44780 classes get_font method to retrieve the internal
+  font used by the device.  This enables us to place exact replicas of the
+  characters within the font tables on the canvas.  When these characters are
+  displayed, because they are already normal characters, they do not consume any
+  of the customer character RAM.  The hd44780 class contains both the A00 and A02
+  font tables.  You should request the table that matches what is installed in
+  your display.
+
+* The progress bar is drawn using normal ImageDraw primitives in this case a
```

Comment:
ImageDraw should link to pillow api docs.

Reply:
Changed as suggested.